xdsl/dialects/llvm.py

```diff
@@ -366,6 +410,96 @@ class LLVMMLIRUndef(IRDLOperation):
     res: Annotated[OpResult, AnyAttr()]
 
 
+@irdl_op_definition
+class GlobalOp(IRDLOperation):
+    name = "llvm.mlir.global"
+
+    global_type: OpAttr[Attribute]
+    constant: OptOpAttr[UnitAttr]
+    sym_name: OpAttr[StringAttr]
+    linkage: OpAttr[LinkageAttr]
+    dso_local: OptOpAttr[UnitAttr]
+    thread_local_: OptOpAttr[UnitAttr]
+    value: OptOpAttr[Attribute]
+    alignment: OptOpAttr[AnyIntegerAttr]
+    addr_space: OpAttr[AnyIntegerAttr]
+    unnamed_addr: OptOpAttr[AnyIntegerAttr]
+    section: OptOpAttr[StringAttr]
+
+    # This always needs an empty region as it is in the top level module definition
+    body: Region
+
+    @staticmethod
+    def get(global_type: Attribute,
+            sym_name: str | StringAttr,
+            linkage: str | LinkageAttr,
+            addr_space: int,
+            constant: bool | None = None,
+            dso_local: bool | None = None,
+            thread_local_: bool | None = None,
+            value: Attribute | None = None,
+            alignment: int | None = None,
+            unnamed_addr: int | None = None,
+            section: str | StringAttr | None = None):
+
+        if isinstance(sym_name, str):
+            sym_name = StringAttr(sym_name)
+
+        if isinstance(linkage, str):
+            linkage = LinkageAttr.from_linkage(linkage)
+
+        attrs: dict[str, Attribute] = {
+            'global_type': global_type,
+            'sym_name': sym_name,
+            'linkage': linkage,
+            'addr_space': IntegerAttr(addr_space, 32)
+        }
+
+        if constant is not None and constant:
+            attrs['constant'] = UnitAttr()
+
+        if dso_local is not None and dso_local:
+            attrs['dso_local'] = UnitAttr()
+
+        if thread_local_ is not None and thread_local_:
+            attrs['thread_local_'] = UnitAttr()
+
+        if value is not None:
+            attrs['value'] = value
+
+        if alignment is not None:
+            attrs['alignment'] = IntegerAttr(alignment, 64)
+
+        if unnamed_addr is not None:
+            attrs['unnamed_addr'] = IntegerAttr(unnamed_addr, 64)
+
+        if section is not None:
+            if isinstance(section, str):
+                section = StringAttr(section)
+            attrs['section'] = section
+
+        return GlobalOp.build(attributes=attrs, regions=[Region([Block()])])
+
+
+@irdl_op_definition
+class AddressOfOp(IRDLOperation):
+    name = "llvm.mlir.addressof"
+
+    global_name: OpAttr[SymbolRefAttr]
+    result: Annotated[OpResult, LLVMPointerType]
+
+    @staticmethod
+    def get(global_name: str | StringAttr | SymbolRefAttr,
```

Comment:
nit: make init instead

Reply:
I hope you don't mind, I am going to keep with get for the moment as this will be consistent with all other operations in the dialect. Later on we can move over to using init throughout the dialect

Reply:
No problem at all, we'll migrate it later